openvdb/openvdb/tree/Tree.h

```diff
@@ -113,11 +113,11 @@ class OPENVDB_API TreeBase
     /// A tree with only a root node and leaf nodes has depth 2, for example.
     virtual Index treeDepth() const = 0;
     /// Return the number of leaf nodes.
-    virtual Index32 leafCount() const = 0;
```

Comment:
Thanks for making all of those changes, I agree with the idea of moving to 64-bit unsigned integers across the whole API so that we can support larger grid sizes. One important thing to note, I believe that these types of change will affect ABI, because it's changing the return type of a virtual function which can (theoretically) lead to a restructuring of the vtable which can cause incompatibilities between different versions of OpenVDB. As a result, I believe it is required that we do an ABI check here and conditionally change this for ABI=12 only. It is good timing to discuss this because we're about the lock the ABI for next year!
```
#if OPENVDB_ABI_VERSION_NUMBER >= 12
virtual Index64 leafCount() const = 0;
#else
virtual Index32 leafCount() const = 0;
#endif
```

Would be good to get thoughts from @kmuseth and/or @jmlait on this one, and I recommend we wait until next week's TSC to finally merge this one.

Reply:
> I agree with the idea of moving to 64-bit unsigned integers across the whole API so that we can support larger grid sizes.

Do you mean the three leafCount-type methods I changed to 64-bit, or do you mean internal node count methods too?

> I recommend we wait until next week's TSC to finally merge this one.

Definitely agree.

Reply:
I believe it's only required for the virtual methods on the Tree classes, but changing non-virtual methods may need logic elsewhere to be adjusted if it could be a mix of Index32 and Index64 for older ABIs.